### DISCUSSION TITLE
SVG ou png?

### DISCUSSION THREAD

**Discussion by silvaMatheus** [frontendbr/forum#791] [Discussão]
Boa noite pessoal achei um desafio em um repositório aonde  pede : 

Aplicação que simula um elevador em JavaScript

utilizando : HTML CSS JavaScript

O Elevador possui 5 andares.
Com botões de chamadas para subir e descer em cada andar.
Abertura das portas e fechamento das portas quando chega ao andar.
voce pode escolher o andar para no painel do elevador.
O elevador tem o tempo de viagem proporcional ao ponto de origem em que ele estava.
Caso alguem chame num andar mais abaixo do que outra chamada acima, ele atende a chamada do andar mais inferior.
Layout responsivo até Iphone 5s.

Ainda estou pensando como irei fazer ja q nao manjo ja q voltei a pegar pesado aos estudos agora mas me auto desafiei a tentar ... Minha duvida é a seguinte, na  questão do layout seria melhor eu pegar uma imagem em png  como essa  e fazer um painel separado com css :
![image](https://user-images.githubusercontent.com/17308187/32205853-d441fcfe-bdd8-11e7-8d21-51802f9301c6.png)

ou fazer todo o layout em SVG? 

obs: Não sei se ficou clara minha ideia desculpem kkkk

**Comment by n3pu** [comment]
Pelo meu entendimento sobre qualidade, vc deveria trabalhar com canvas. Se optar por svg seria pela facilidade e qualidade do resultado. Como tem uma condição diferente de tempo pra cada ocasião então vai ter q trabalhar com canvas. Imagino isso =)

**Comment by brunoravanhani** [comment]
Cara, se puder (e tiver tempo) use SVG, vai te dar mais liberdade e é mais leve. Seria a minha opção no caso.

**Comment by deppbrazil** [comment]
**SVG** fica mais leve pois é tudo codigo, porém vc teria que vetorizar tudo no **Ai** **(illustrator)** ou algum software similar e exportar em **SVG**, mas o Ai gera um **SVG** com um codigo muito sujo no qual vc perde um pouco de performance principalmente se vc usar os efeitos de **3D** e rotação nativos do **CSS** vai notar que o **SVG** vai renderizar meio embaçado no browser **(pelo menos tive um problema parecido)** e fazer o **SVG** na mão vai te levar tempo.... 

### Não sei se alguém usa outra técnica na produção de **SVG´s**! se tiver pinga ai =)

Como é um teste que pede **HTML**, **CSS** e **JavaScript** creio que eles estão avaliando a funcionalidade e não tanto se vc usar **PNG** ou **SVG**... pq ate mesmo se vc usar os **PNG** otimizados para **Web** **(medias em px, e 70 a 100dpi na qualidade)** a performance tb fica ótima e te economiza um tempo do `c@6%¨8($##`... rsrs

## Regards! =) ##

**Comment by kivervinicius** [comment]
Desenha tudo com alguma ferramenta vetorial (recomendo Inkscape) e gera arquivos separados e passa o [SVGO](https://github.com/svg/svgo) para limpar todas tags. Com isso você pode ter mais liberdade na hora de fazer a página.

**Comment by silvaMatheus** [comment]
vlw pessoal pelo visto SVG é uma das melhores opções, ja vetorizei um desenho q eu fiz 
![image](https://user-images.githubusercontent.com/17308187/32237223-8df27242-be4b-11e7-937c-2f4b3f136098.png) agora irei estudar como trabalhar as animações muito obg mesmo

**Comment by bernardodiasc** [comment]
Show @silvaMatheus 👍

**Comment by ranosi** [comment]
Tu tem como disponibilizar esse desafio pra galera também?
Em ter, 31 de out de 2017 às 23:31, Bernardo Dias <notifications@github.com>
escreveu:
